float32 current_x
float32 current_y
bool navigating
